ga4gh/cli.py

```diff
@@ -1510,448 +1521,595 @@ def configtest_main(parser=None):
+        # TODO this is _very_ crude. We need much more options and detail here.
+        self._repo.printSummary()
 
+    def verify(self):
+        """
+        Checks that the data pointed to in the repository works and
+        we don't have any broken URLs, missing files, etc.
+        """
+        self._openRepo()
+        self._repo.verify()
 
-class RemoveReferenceSetRunner(AbstractRepoCommandRunner):
+    def addOntology(self):
+        """
+        Adds a new Ontology to this repo.
+        """
+        self._openRepo()
+        name = self._args.name
+        if name is None:
+            name = getNameFromPath(self._args.filePath)
+        ontologyTermMap = ontologies.OntologyTermMap(name)
+        ontologyTermMap.populateFromFile(self._args.filePath)
+        self._updateRepo(self._repo.insertOntologyTermMap, ontologyTermMap)
+
+    def addDataset(self):
+        """
+        Adds a new dataset into this repo.
+        """
+        self._openRepo()
+        dataset = datasets.Dataset(self._args.datasetName)
+        dataset.setDescription(self._args.description)
+        self._updateRepo(self._repo.insertDataset, dataset)
 
-    def __init__(self, args):
-        super(RemoveReferenceSetRunner, self).__init__(args)
-        self.referenceSetName = args.referenceSetName
+    def addReferenceSet(self):
+        """
+        Adds a new reference set into this repo.
+        """
+        self._openRepo()
+        name = self._args.name
+        if name is None:
+            name = getNameFromPath(self._args.filePath)
+        referenceSet = references.HtslibReferenceSet(name)
+        referenceSet.populateFromFile(self._args.filePath)
+        referenceSet.setDescription(self._args.description)
+        # TODO other reference set metadata.
+        self._updateRepo(self._repo.insertReferenceSet, referenceSet)
+
+    def addReadGroupSet(self):
+        """
+        Adds a new ReadGroupSet into this repo.
+        """
+        self._openRepo()
+        dataset = self._repo.getDatasetByName(self._args.datasetName)
+        dataUrl = self._args.dataFile
+        indexFile = self._args.indexFile
+        parsed = urlparse.urlparse(dataUrl)
+        # TODO, add https support and others when they have been
+        # tested.
+        if parsed.scheme in ['http', 'ftp']:
+            if indexFile is None:
+                raise exceptions.MissingIndexException(dataUrl)
+        else:
+            if indexFile is None:
+                indexFile = dataUrl + ".bai"
+        name = self._args.name
+        if self._args.name is None:
+            name = getNameFromPath(dataUrl)
+        readGroupSet = reads.HtslibReadGroupSet(dataset, name)
+        readGroupSet.populateFromFile(dataUrl, indexFile)
+        referenceSetName = self._args.referenceSetName
+        if referenceSetName is None:
+            # Try to find a reference set name from the BAM header.
+            referenceSetName = readGroupSet.getBamHeaderReferenceSetName()
```

Comment:
Because we need to create a linkage between readgroupsets and referencesets. There is a ReferenceSet field in ReadGroupSet which we must fill. If deriving the reference set from the BAM header proves to be problematic we can make it a mandatory argument.